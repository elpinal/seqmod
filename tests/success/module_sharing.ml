{
  module M = {
    type t = unit
    val x = ()
  }

  signature S = like M

  signature T = {
    module N : {type t}
    module O = N
  }
}
