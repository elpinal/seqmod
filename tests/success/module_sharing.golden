purity: pure
∃!5.{M:{x(=!5):unit, t(=unit):Type}, S={x(=!5):unit, t(=unit):Type}, T=∃!6.{N:{t(=!6):Type}, O:{t(=!6):Type}}}
